### PR TITLE
Realm App Importer: Environment variables & API keys

### LINKED ISSUE #4154: Extend realm-app-importer to read baseUrl from an environment variable & support API key authentication
We need this to enable running integration tests against a server from GitHub Actions.

### PR DESCRIPTION
**Pull request by kraenhansen**
## What, How & Why?

This close #4154 by adding to the Realm App Importer:
- Possibility to authenticate using API keys
- Specify base URL, username, password, public and private keys via environment variables.

## ☑️ ToDos
* [ ] 📝 Changelog entry
* [x] 📝 `Compatibility` label is updated or copied from previous entry
* [x] 🚦 Tests (locally & manually)
* [x] 📱 Check the React Native/other sample apps work if necessary
* [x] 📝 Public documentation PR created or is not necessary
* [x] 💥 `Breaking` label has been applied or is not necessary

### CODE REVIEW

**Review by kneth — APPROVED** [review]

File: packages/realm-app-importer/src/AppImporter.ts [hunk 1]
```diff
@@ -182,22 +191,50 @@ export class AppImporter {
     cp.execFileSync(cliPath, args, { stdio: "inherit" });
   }
 
+  private performLogIn() {
+    const { credentials } = this;
+    console.log(credentials);
```
Comment (takameyer) [inline packages/realm-app-importer/src/AppImporter.ts:196]: Probably don't want to log this

File: packages/realm-app-importer/src/AppImporter.ts [hunk 2]
```diff
@@ -182,22 +191,50 @@ export class AppImporter {
...
+    if (credentials.kind === "api-key") {
+      return fetch(`${this.apiUrl}/auth/providers/mongodb-cloud/login`, {
+        method: "POST",
+        headers: { "content-type": "application/json" },
+        body: JSON.stringify({
+          username: credentials.publicKey,
```
Comment (takameyer) [inline packages/realm-app-importer/src/AppImporter.ts:202]: Is `username` the correct field name?  I don't know the API, just wanted to make sure this wasn't a typo, as this name doesn't really reflect what it is.

File: packages/realm-app-importer/src/cli.ts [hunk 3]
```diff
@@ -92,6 +92,36 @@ function resolveAppTemplates(paths: string[]): Record<string, string> {
...
+  username: process.env.REALM_USERNAME || "unique_user@domain.com",
+  password: process.env.REALM_PASSWORD || "password",
+  publicKey: process.env.REALM_PUBLIC_KEY,
+  privateKey: process.env.REALM_PRIVATE_KEY,
+};
+
+type CredentialsOptions = {
+  username: string;
+  password: string;
+  publicKey: string | undefined;
```
Comment (takameyer) [inline packages/realm-app-importer/src/cli.ts:106]: Any reason to not use `publicKey?: string` ?

File: packages/realm-app-importer/tsconfig.build.json [hunk 4]
```diff
@@ -1,9 +1,9 @@
 {
 	"extends": "./tsconfig.json",
 	"compilerOptions": {
-		"target": "ES5",
+		"target": "es2018",
```
Comment (takameyer) [inline packages/realm-app-importer/tsconfig.build.json:4]: Can we go straight for `es2020`?

**Review by takameyer — COMMENTED** [review]
Got to this late, but have a few comments.

File: packages/realm-app-importer/src/AppImporter.ts (hunk 2 above)
Comment (kraenhansen) [reply packages/realm-app-importer/src/AppImporter.ts:202]: Yeah - it's kind of weird that it's sending the public key like that, but I did check the `realm-cli` package: https://github.com/10gen/realm-cli/blob/master/internal/cloud/realm/auth.go#L22-L25

File: packages/realm-app-importer/src/cli.ts (hunk 3 above)
Comment (kraenhansen) [reply packages/realm-app-importer/src/cli.ts:106]: Basically, `username` and `password` has default values and the `publicKey` and `privateKey` doesn't.

File: packages/realm-app-importer/tsconfig.build.json (hunk 4 above)
Comment (kraenhansen) [reply packages/realm-app-importer/tsconfig.build.json:4]: Most likely, but I didn't have to for now.

**Pull request merged**